src/sharepoint/rest/rest.ts

```diff
@@ -1,10 +1,75 @@
 "use strict";
 
-import { Web } from "./web";
+import { Site } from "./site";
+import { Web } from "./webs";
+import * as Util from "../../utils/util";
+import { Queryable } from "./queryable";
 
 /**
  * Root of the SharePoint REST module
  */
 export class Rest {
-    public web = new Web("_api");
+
+    /**
+     * Begins a site collection scoped REST request
+     * 
+     * @param url The base url for the request, optional if running in the context of a page
+     */
+    public get site(): Site {
+        return new Site("_api");
+    }
+
+    /**
+     * Begins a web scoped REST request
+     * 
+     * @param url The base url for the request, optional if running in the context of a page
+     */
+    public get web(): Web {
+        return new Web("_api");
+    }
+
+    /**
+     * Begins a cross-domain, host site scoped REST request, for use in add-in webs
+     * 
+     * @param addInWebUrl The absolute url of the add-in web
+     * @param hostWebUrl The absolute url of the host web
+     */
+    public crossDomainSite(addInWebUrl: string, hostWebUrl: string): Site {
+        return this._cdImpl<Site>(Site, addInWebUrl, hostWebUrl, "site");
+    }
+
+    /**
+     * Begins a cross-domain, host web scoped REST request, for use in add-in webs
+     * 
+     * @param addInWebUrl The absolute url of the add-in web
+     * @param hostWebUrl The absolute url of the host web
+     */
+    public crossDomainWeb(addInWebUrl: string, hostWebUrl: string): Web {
```

Comment:
We discussed this on one of the calls, either last week or two weeks ago - the idea of ambient configuration. My issue with this approach is it then becomes a little "magical" in the sense that if you go fiddle with these settings and such then this starts happening. If we can come up with a clean way to do it I am not against it.

What I did makes the call explicit, and for 90% of the folks they will never use it and continue to just use .web.* as before.